packages/orchestrator/internal/service/machineinfo/main.go

```diff
@@ -0,0 +1,31 @@
+package machineinfo
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/shirou/gopsutil/v4/cpu"
+)
+
+type MachineInfo struct {
+	Family string
+	Model  string
+	Arch   string
+}
+
+func Detect() (MachineInfo, error) {
+	info, err := cpu.Info()
```

Comment:
Will this work on both AWS and GCP instances? At least for AWS they have Graviton processors that are not explicitly mentioned here https://github.com/shirou/gopsutil/blob/master/cpu/cpu_linux.go